File: xadditus-android/app/src/main/java/com/linoagli/android/xadditus/base/BaseActivity.java
/**
 * xadditus-android Project.
 * com.linoagli.android.xadditus.base
 *
 * @author Faye-Lino Agli, username: linoagli
 */
package com.linoagli.android.xadditus.base;

import android.support.v7.app.AppCompatActivity;
import com.linoagli.android.xadditus.AppRuntime;
import org.greenrobot.eventbus.EventBus;

abstract public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onResume() {
        super.onResume();
        AppRuntime.isAppInForeground = true;
        EventBus.getDefault().register(this);
    }

    @Override
    protected void onPause() {
        EventBus.getDefault().unregister(this);
        AppRuntime.isAppInForeground = false;
        super.onPause();
    }
}

File: xadditus-android/app/src/main/kotlin/com/linoagli/android/xadditus/base/BaseActivity.kt
/**
 * xadditus-android Project.
 * com.linoagli.android.xadditus.base
 * @author Faye-Lino Agli, username: linoagli
 */
package com.linoagli.android.xadditus.base

import android.support.v7.app.AppCompatActivity
import com.linoagli.android.xadditus.Runtime
import org.greenrobot.eventbus.EventBus

abstract class BaseActivity : AppCompatActivity() {
    override fun onResume() {
        super.onResume()
        Runtime.isAppInForeground = true
        EventBus.getDefault().register(this)
    }

    override fun onPause() {
        EventBus.getDefault().unregister(this)
        Runtime.isAppInForeground = false
        super.onPause()
    }
}